code/datums/traits.dm

```diff
@@ -205,8 +205,9 @@
 	proc/getTraitWithCategory(var/cat)
 		for(var/id in traits)
 			var/datum/trait/T = traits[id]
-			if (T.category == cat)
-				. = T
+			for (var/heldcat in T.category)
+				if (heldcat == cat)
+					. = T
```

Comment:
Any reason why this isn't just `return T`? It'd be a little bit faster and just generally cleaner code.

Reply:
Nope-- This is just how it was before and I generally try to keep code the same as it was before since my understanding is pretty poor... I'll do a test run with it as "return" to see if it works and update it if it does